src/main/java/com/waverleysoftware/pageobjects/BasePage.java

```diff
@@ -0,0 +1,56 @@
+package com.waverleysoftware.pageobjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+import static org.openqa.selenium.support.ui.ExpectedConditions.stalenessOf;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
+
+/**
+ * A simple base class for processing common methods for all children classes.
+ */
+public class BasePage {
+
+    protected final WebDriver driver;
+    protected final WebDriverWait wait;
+
+    public BasePage(final WebDriver driver) {
+        this.wait = new WebDriverWait(driver, 10);
+        this.driver = driver;
+    }
+
+    public void click(final By locator) {
+        find(locator).click();
+    }
+
+    public void type(final By locator, final String text) {
+        click(locator);
```

Comment:
why do you need to click input before interaction?.. it's automatically focused by selenium